src/workers/build/mod.rs

```diff
@@ -0,0 +1,151 @@
+    match project.project_type {
+        ProjectType::JavaScript => {
+            message::info("JavaScript project found. Skipping unnecessary build!")
+        }
+        ProjectType::Rust => {
+            let tool_name = "wasm-pack";
+            let binary_path = install::install(tool_name, "rustwasm")?.binary(tool_name)?;
+            let args = ["build", "--target", "no-modules"];
+
+            let command = command(&args, binary_path);
+            let command_name = format!("{:?}", command);
+
+            commands::run(command, &command_name)?;
+        }
+        ProjectType::Webpack => {
+            wranglerjs::run_build(project)?;
+        }
+    }
+
+    let worker = match project.project_type {
+        ProjectType::Rust => wasm_worker(&project.name)?,
+        ProjectType::Webpack => webpack_worker(&project.name)?,
+        ProjectType::JavaScript => js_worker(&project.name)?,
+    };
+    // add other resoureces
+    Ok(worker)
+}
+
+fn wasm_worker(project_name: &str) -> Result<Worker, failure::Error> {
+    let name = krate::Krate::new("./")?.name.replace("-", "_");
+
+    build_generated_dir()?;
+    concat_js(&name)?;
+
+    let script_path = "./worker/generated/script.js";
+    let wasm_path = format!("./pkg/{}_bg.wasm", name);
+
+    let wasm_resource = wasm_resource(wasm_path)?;
+
+    Ok(Worker {
+        name: project_name.to_string(),
+        script: Script {
+            name: "script".to_string(),
+            path: script_path.to_string(),
+        },
+        resources: vec![wasm_resource],
+    })
+}
+
+fn webpack_worker(project_name: &str) -> Result<Worker, failure::Error> {
+    let script_path = "./worker/script.js";
+    let mut worker = Worker {
+        name: project_name.to_string(),
+        script: Script {
+            name: "script".to_string(),
+            path: script_path.to_string(),
+        },
+        resources: Vec::new(),
+    };
+    let wasm_path = "./worker/module.wasm";
+    if Path::new(wasm_path).exists() {
+        let wasm_resource = wasm_resource(wasm_path.to_string())?;
+        worker.resources.push(wasm_resource);
+    };
+    Ok(worker)
+}
+
+fn js_worker(project_name: &str) -> Result<Worker, failure::Error> {
+    let pkg = Package::new("./")?;
+    let script_path = pkg.main()?;
+    Ok(Worker {
+        name: project_name.to_string(),
+        script: Script {
+            name: "script".to_string(),
+            path: script_path,
+        },
+        resources: Vec::new(),
+    })
+}
+
+fn wasm_resource(wasm_path: String) -> Result<Resource, failure::Error> {
+    let metadata_path = "./worker/metadata_wasm.json";
+    let mut binding = "wasm".to_string();
+
+    if Path::new(metadata_path).exists() {
+        let file = fs::File::open(metadata_path)?;
+        let reader = BufReader::new(file);
+        let data: serde_json::Value = serde_json::from_reader(reader)?;
+        binding = match data["bindings"][0]["name"].as_str() {
+            Some(s) => s.to_string(),
```

Comment:
i would probably just use ? here for now

Reply:
Wouldn't I have to implement `From<NoneError>`? is that possible? I don't own the definition of `NoneError` or `failure::Error`, so I don't think I can implement either `From` or `Into` for this. Maybe I'm misunderstanding something.